src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -273,18 +273,31 @@ public override Task<bool> HandleRequestAsync()
         /// Response to the callback from OpenId provider after session ended.
         /// </summary>
         /// <returns>A task executing the callback procedure</returns>
-        protected virtual Task<bool> HandleSignOutCallbackAsync()
+        protected async virtual Task<bool> HandleSignOutCallbackAsync()
         {
-            if (Request.Query.TryGetValue(OpenIdConnectParameterNames.State, out StringValues protectedState))
+            Request.Query.TryGetValue(OpenIdConnectParameterNames.State, out StringValues protectedState);
+            var properties = Options.StateDataFormat.Unprotect(protectedState);
+
+            var redirectContext = new SignedOutRedirectContext(Context, Scheme, Options, properties);
```

Comment:
To make custom parameters easier to retrieve from the OIDC post logout notification request, maybe it would be worth instantiating and flowing an `OpenIdConnectMessage` built from the query string parameters?